samza-core/src/main/scala/org/apache/samza/checkpoint/CheckpointTool.scala

```diff
@@ -152,25 +153,34 @@ object CheckpointTool {
     val cmdline = new CheckpointToolCommandLine
     val options = cmdline.parser.parse(args: _*)
     val config = cmdline.loadConfig(options)
-    val rconfig = rewriteConfig(new JobConfig(config))
-    print("Rewritten config" + rconfig)
-    val tool = CheckpointTool(rconfig, cmdline.newOffsets)
-    tool.run
+    val rewrittenConfig = rewriteConfig(new JobConfig(config))
+    info(s"Using the rewritten config: $rewrittenConfig")
+    val tool = CheckpointTool(rewrittenConfig, cmdline.newOffsets)
+    tool.run()
   }
 }
 
-class CheckpointTool(config: Config, newOffsets: TaskNameToCheckpointMap, manager: CheckpointManager) extends Logging {
+class CheckpointTool(newOffsets: TaskNameToCheckpointMap, coordinatorStreamManager: CoordinatorStreamManager) extends Logging {
+
+  def run() {
+    // Read the configuration stored in the coordinator stream.
+    coordinatorStreamManager.register(getClass.getSimpleName)
+    coordinatorStreamManager.start()
+    coordinatorStreamManager.bootstrap()
+    val coordinatorStreamConfig: Config = coordinatorStreamManager.getConfig
 
-  def run {
-    info("Using %s" format manager)
+    // Instantiate the checkpoint manager with coordinator stream configuration.
+    val checkpointManager: CheckpointManager = coordinatorStreamConfig.getCheckpointManagerFactory() match {
+      case Some(className) =>
+        Util.getObj(className, classOf[CheckpointManagerFactory])
+          .getCheckpointManager(coordinatorStreamConfig, new MetricsRegistryMap)
```

Comment:
Is this being moved to `run` because `CheckpointManagerFactory` now needs the full Config to build a checkpoint manager?

Reply:
Yes, to instantiate CheckpointManager with full config.